### PR TITLE
DAOS-11309 Build Failure: Building of java jar files as mentioned in …

### PR DESCRIPTION
**Pull request by zjf2012**
…documentation not working

put back missing assemble xml file and daos-site.xml. And renaming daos-site.xml to core-site-daos-ref.xml which is to be merged to Hadoop core-site.xml since we don't want to add new configuration file to hadoop for daos.

updated documentation accordingly and remove transitive deps which has security vulnerability.

Signed-off-by: jiafu zhang <jiafu.zhang@intel.com>

### CODE REVIEW

**Comment by zjf2012** [comment]
@johannlombardi  @saurabhtandan , the failed jobs seem not be related to my changes. please help review.

File: docs/user/spark.md [hunk 1]
```diff
@@ -57,49 +60,31 @@ Export all DAOS-specific env variables in your application, e.g.,
 `spark-env.sh` for Spark and `hadoop-env.sh` for Hadoop. Or you can simply put
 env variables in your `.bashrc`.
 
-Besides, you should have LD\_LIBRARY\_PATH include DAOS library path so that Java
-can link to DAOS libs, like below.
+Besides, you should have `LD_LIBRARY_PATH` include DAOS library path so that
+Java can link to DAOS libs, like below.
```
Comment (johannlombardi) [inline docs/user/spark.md:64]: Is that needed if installed from RPM?

**Review by johannlombardi — APPROVED** [review]

File: docs/user/spark.md (hunk 1 above)
Comment (zjf2012) [reply docs/user/spark.md:64]: good catch. let me rephrase it.

**Comment by johannlombardi** [comment]
@ashleypittman This PR is failing NLT in docker with:

Error: No such container:path: build-post:/home/daos/daos/nlt-junit.xml
[10](https://github.com/daos-stack/daos/runs/7943121888?check_suite_focus=true#step:8:11)
Error: Process completed with exit code 1.

Any thoughts? thanks in advance.

**Comment by ashleypittman** [comment]
Build logs says 'docker: Error response from daemon: failed to create shim: OCI runtime create failed: container_linux.go:380: starting container process caused: exec: "./daos/utils/ci/run_in_gha.sh": stat ./daos/utils/ci/run_in_gha.sh: no such file or directory: unknown.'

We just renamed that file so I think this may be an issue where github actions runs with the trigger/workflow from the target branch but the code from the PR and that update that change the interface between the two cause problems.

It should resolve itself if this is merged with master.

**Review by johannlombardi — APPROVED** [review]

**Comment by johannlombardi** [comment]
Just merged master.

File: docs/user/spark.md [hunk 2]
```diff
@@ -37,17 +38,19 @@ see [Build DAOS Hadoop Filesystem](#builddaos) for details.
...
 
-### `daos-site-example.xml`
+### `core-site-daos-ref.xml` (version >= 2.2.1, or `daos-site-example.xml`)
 
-Extract from `hadoop-daos-<version>.jar` and rename to `daos-site.xml`. Then
-copy it to your application config directory, e.g., `$SPARK_HOME/conf` for
-Spark and `$HADOOP_HOME/etc/hadoop` for Hadoop.
+Extract from `hadoop-daos-<version>-protobuf3-netty4-shaded.jar`. Then merge
+with your Hadoop `core-site.xml` which is under your application config
+directory, e.g., `$SPARK_HOME/conf` for Spark and `$HADOOP_HOME/etc/hadoop`
```
Comment (saurabhtandan) [inline docs/user/spark.md:52]: Is there supposed to be a core-site.xml file under `$SPARK_HOME/conf`. My Spark download originally did not have any `core-site.xml` which I can merge with `core-site-daos-ref.xml`. I do see it for Hadoop under `$HADOOP_HOME/etc/hadoop` button for Spark.

File: docs/user/spark.md [hunk 3]
```diff
@@ -128,9 +114,9 @@ spark.driver.extraClassPath     /path/to/daos-java-<version>.jar:/path/to/hadoop
 ### Validatng Spark Access
```
Comment (saurabhtandan) [inline docs/user/spark.md:114]: Typo:
s/Validatng/Validating

**Review by saurabhtandan — CHANGES_REQUESTED** [review]
The jar build failed originally for me with the following error message:
`Execution default of goal org.apache.maven.plugins:maven-assembly-plugin:2.6:single failed: group id '11579472' is too big ( > 2097151 ). Use STAR or POSIX extensions to overcome this limit`
And I had to go inside `distribution/pom.xml` and add the following line `<tarLongFileMode>posix</tarLongFileMode>` for maven-assembly-plugin block to get a successful build. 
Is there a permanent solution we can have in the code to resolve this issue?

**Comment by johannlombardi** [comment]
@zjf2012 any thoughts on the above?

File: docs/user/spark.md (hunk 2 above)
Comment (zjf2012) [reply docs/user/spark.md:52]: Good catch.

Spark can work with or without Hadoop installation. If it's with Hadoop inst., you only need to merge DAOS related properties into $HADOOP_HOME/etc/hadoop/core-site.xml. If it's without Hadoop inst., you have two ways to put your core-site.xml. One is to put it in $SPARK_HOME/conf. The other is to put the directory to which your HADOOP_CONF_DIR env points.

I need to rephrase it. thanks.

File: docs/user/spark.md (hunk 3 above)
Comment (zjf2012) [reply docs/user/spark.md:114]: good catch.

**Comment by zjf2012** [comment]
> The jar build failed originally for me with the following error message: `Execution default of goal org.apache.maven.plugins:maven-assembly-plugin:2.6:single failed: group id '11579472' is too big ( > 2097151 ). Use STAR or POSIX extensions to overcome this limit` And I had to go inside `distribution/pom.xml` and add the following line `<tarLongFileMode>posix</tarLongFileMode>` for maven-assembly-plugin block to get a successful build. Is there a permanent solution we can have in the code to resolve this issue?

I think it's OS specific issue. For now, I am afraid the best solution your temp workaround. let me add it to the pom file.

**Review by johannlombardi — DISMISSED** [review]

File: src/client/java/distribution/pom.xml [hunk 4]
```diff
@@ -30,6 +30,7 @@
               <descriptors>
                 <descriptor>src/assembly/make-assembly.xml</descriptor>
               </descriptors>
+	      <tarLongFileMode>posix</tarLongFileMode>
```
Comment (saurabhtandan) [inline src/client/java/distribution/pom.xml:33]: Building the jar files is still failing. There are 3 different sections in this pom.xml
1. <descriptor>src/assembly/make-assembly.xml</descriptor>
2. <descriptor>src/assembly/make-assembly-with-dependencies.xml</descriptor>
3. <descriptor>src/assembly/make-assembly-with-proto3-netty4-deps.xml</descriptor>

The above change is for section 1. But `<tarLongFileMode>posix</tarLongFileMode>` should be added to those other 2 sections of pom.xml as well.
Specially for section 3 as we recommend building with `with-proto3-netty4-deps` in documentation i.e
`mvn -Pdistribute,with-proto3-netty4-deps clean package -DskipTests -Dgpg.skip -Ddaos.install.path=<daos_install>`

Comment (saurabhtandan) [reply src/client/java/distribution/pom.xml:33]: Add `<tarLongFileMode>posix</tarLongFileMode>` after following lines as well:

https://github.com/daos-stack/daos/blob/c3bee59024a53a97a7943bf09a70c0237fbcaa7c/src/client/java/distribution/pom.xml#L58

https://github.com/daos-stack/daos/blob/c3bee59024a53a97a7943bf09a70c0237fbcaa7c/src/client/java/distribution/pom.xml#L83

Comment (zjf2012) [reply src/client/java/distribution/pom.xml:33]: done

**Review by johannlombardi — APPROVED** [review]

**Review by saurabhtandan — APPROVED** [review]
LGTM
Thanks Jiafu.

**Pull request merged**